onnx/defs/nn/defs.cc

```diff
@@ -184,6 +184,98 @@ void convPoolShapeInference(
   }
 }
 
+void imageToColShapeInference(
+    InferenceContext& ctx) {
+  // we need the first input shape for this inference.
+  if (!hasInputShape(ctx, 0)) {
+    return;
+  }
+
+  auto input_shape = ctx.getInputType(0)->tensor_type().shape();
+  if (input_shape.dim_size() < 2) {
+    fail_shape_inference("Input tensor must have at least 2 dimensions");
+  }
+
+  // first dim is the batch axis and the next is the number of channels.
+  size_t n_input_dims = static_cast<size_t>(input_shape.dim_size() - 2);
+
+  std::vector<int64_t> dilations;
+  if (getRepeatedAttribute(ctx, "dilations", dilations)) {
+    if (dilations.size() != n_input_dims) {
+      fail_shape_inference("Attribute dilations has incorrect size");
+    }
+  } else {
+    dilations.assign(n_input_dims, 1);
+  }
+
+  std::vector<int64_t> strides;
+  if (getRepeatedAttribute(ctx, "strides", strides)) {
+    if (strides.size() != n_input_dims) {
+      fail_shape_inference("Attribute strides has incorrect size");
+    }
+  } else {
+    strides.assign(n_input_dims, 1);
+  }
+
+  std::vector<int64_t> block_shape;
+  if (getRepeatedAttribute(ctx, "block_shape", block_shape)) {
+    if (block_shape.size() != n_input_dims) {
+      fail_shape_inference("Attribute size has incorrect size");
+    }
+  } else {
+    fail_shape_inference("Attribute block_shape must be specified");
+  }
+
+  size_t block_num_element = input_shape.dim(1).dim_value();
```

Comment:
I think this also needs a has_dim_value check: if it is not known, the corresponding output-dim should be left undefined.